apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerMiddleware.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// Middleware that will, when invoked, call the next <see cref="RequestDelegate"/>,
+    /// catch any exception that it may throw, send that exception to the
+    /// Stackdriver Error Reporting API and rethrow the given exception.
+    /// </summary>
+    public sealed class ErrorReportingExceptionLoggerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ErrorReportingExceptionLogger _logger;
+
+        /// <summary>
+        /// Create a new instance of <see cref="ErrorReportingExceptionLoggerMiddleware"/>.
+        /// </summary>
+        /// <param name="next">The next request delegate.</param>
+        /// <param name="logger">A logger that will report exceptions to the Stackdriver
+        ///     Error Reporting API.</param>
+        public ErrorReportingExceptionLoggerMiddleware(RequestDelegate next, ErrorReportingExceptionLogger logger)
+        {
+            _next = next;
```

Comment:
Is this allowed to be null? It is used below assuming it is non-null, so it might be good to check it here. Or if it can be null, use `_next?.Invoke(httpContext)` below.